Update service-fabric-reliable-services-reliable-collections-guidelin…

plutoscarab opened the pull request:
…es.md

Clarify that deep copying of structs may be necessary because structs can contain non-struct members.

azurecla commented:
Hi __@BrucePerlerMS__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        This seems like a small (but important) contribution, so no contribution license agreement is required at this point. Real humans will now evaluate your PR.
    </span>

TTYL, AZPRBOT;

plutoscarab committed:
Update service-fabric-reliable-services-reliable-collections-guidelines.md

Clarify that deep copying of structs may be necessary because structs can contain non-struct members.
